src/riakc_pb_socket.erl

```diff
@@ -2382,14 +2415,58 @@ process_response(#request{msg = #rpbfetchreq{}},
                  #rpbfetchresp{deleted = true, 
                                 crc_check = CRC,
                                 replencoded_object = ObjBin,
-                                deleted_vclock = VclockBin}, State) ->
+                                deleted_vclock = VclockBin,
+                                segment_id = undefined,
+                                segment_hash = undefined}, State) ->
     {reply,
         {crc_check(CRC,ObjBin), {deleted, VclockBin, ObjBin}},
         State};
+process_response(#request{msg = #rpbfetchreq{}},
+                 #rpbfetchresp{deleted = true, 
+                                crc_check = CRC,
+                                replencoded_object = ObjBin,
+                                deleted_vclock = VclockBin,
+                                segment_id = SegID,
+                                segment_hash = SegHash}, State) ->
+    {reply,
+        {crc_check(CRC,ObjBin), {deleted, VclockBin, ObjBin, SegID, SegHash}},
+        State};
 process_response(#request{msg = #rpbfetchreq{}},
                  #rpbfetchresp{crc_check = CRC,
-                                replencoded_object = ObjBin}, State) ->
+                                replencoded_object = ObjBin,
+                                segment_id = undefined,
+                                segment_hash = undefined}, State) ->
     {reply, {crc_check(CRC,ObjBin), ObjBin}, State};
+process_response(#request{msg = #rpbfetchreq{}},
+                 #rpbfetchresp{crc_check = CRC,
+                                replencoded_object = ObjBin,
+                                segment_id = SegID,
+                                segment_hash = SegHash}, State) ->
+    {reply, {crc_check(CRC,ObjBin), {ObjBin, SegID, SegHash}}, State};
+
+%% rpbpushreq
+process_response(#request{msg = #rpbpushreq{queuename = Q}}, 
+                    #rpbpushresp{queue_exists = true,
+                                    queuename = Q,
+                                    foldq_length = FL,
+                                    fsync_length = FSL,
+                                    realt_length = RTL}, State) ->
+    {reply,
+        {ok, 
+            list_to_binary(
+                lists:flatten(
+                    io_lib:format("Queue ~s: ~w ~w ~w", [Q, FL, FSL, RTL])))},
+        State};
+process_response(#request{msg = #rpbpushreq{queuename = Q}}, 
+                    #rpbpushresp{queue_exists = false}, State) ->
+    {reply,
+        {ok, 
+            list_to_binary(
```

Comment:
```suggestion
            iolist_to_binary(
```
then you can leave out flatten... same in line 2456

Reply:
Fixed